presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -271,33 +289,39 @@ public PinotQueryGeneratorContext visitProject(ProjectNode node, PinotQueryGener
             PinotQueryGeneratorContext context = node.getSource().accept(this, contextIn);
             requireNonNull(context, "context is null");
             LinkedHashMap<VariableReferenceExpression, Selection> newSelections = new LinkedHashMap<>();
-
+            LinkedHashSet<VariableReferenceExpression> newOutputs = new LinkedHashSet<>();
             node.getOutputVariables().forEach(variable -> {
                 RowExpression expression = node.getAssignments().get(variable);
                 PinotExpression pinotExpression = expression.accept(
                         contextIn.getVariablesInAggregation().contains(variable) ?
-                                new PinotAggregationProjectConverter(typeManager, functionMetadataManager, standardFunctionResolution, session) : pinotProjectExpressionConverter,
+                        new PinotAggregationProjectConverter(typeManager, functionMetadataManager, standardFunctionResolution, session) : pinotProjectExpressionConverter,
                         context.getSelections());
                 newSelections.put(
                         variable,
                         new Selection(pinotExpression.getDefinition(), pinotExpression.getOrigin()));
+                newOutputs.add(variable);
             });
-            return context.withProject(newSelections);
+            if (useSqlSyntax) {
```

Comment:
I still don't think I am understanding why this is done :-(. 

The project node gives us the projections it wants, and we create `newSelections` (and `newOutputs`) out of that. So if we use the SQL syntax, why are we retaining what the old selections were ? A comment with some example query/plan when this is needed would certainly help !

If that's the case (that old selections should be retained): Why is not done unconditionally ? Why is it not done by the context ?

In addition, should `selections` now be a simple HashMap instead of a LinkedHashMap ? There is a new structure `outputs` to maintain the ordering now.

Reply:
The reason is `newSelections` contains the mapping from `varRef` to `rowExpression`.
`output`, `groupby`, `orderby` will hold only `varRef`.

When we try to generate pinot query, the `varRef` in `groupby` may not be in `output`, e.g. `select count(*) group by A`, in PQL, we expect `A` is always in `selections`. 
Here in SQL, we need to hold all the mappings somewhere, which is in selections.

Since I somehow don't want to change the logic to generate PQL, I prefer to keep the logic of selection as is

Reply:
So in theory, can we do this: Make `selections` be a simple map (not a linked hash map) from `varRef` to `rowExpression`. And `output` is the linked hash set of `varRef` that says exactly what is actually projected out. And do this for both PQL and SQL endpoints.

ie, in your example above: Both `A` and `count(*)` will be in `selections`, while output will only consist of `count(*)`.  Would that work ?

Reply:
this works for SQL.
For PQL, we still need to figure out the mapping of expected column indices based on group by conventions (PQL always return groupbys first then aggregations).